tiledb/sm/c_api/tiledb.h

```diff
@@ -4636,32 +4626,64 @@ TILEDB_EXPORT int32_t tiledb_query_condition_alloc(
  */
 TILEDB_EXPORT void tiledb_query_condition_free(tiledb_query_condition_t** cond);
 
+/**
+ * Initializes a TileDB query condition object.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_query_condition_t* query_condition;
+ * tiledb_query_condition_alloc(ctx, &query_condition);
+ *
+ * uint32_t value = 5;
+ * tiledb_query_condition_init(
+ *   ctx, query_condition, "longitude", &value, sizeof(value), TILEDB_LT);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param cond The allocated query condition object.
+ * @param attribute_name The attribute name.
+ * @param condition_value The value to compare against an attribute value.
+ * @param condition_value_size The byte size of `condition_value`.
+ * @param op The comparison operator.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_query_condition_init(
```

Comment:
I am fine with adding an extra API function. However, shall we call this differently? What name would we use for the "is null" or "is not null" cases we discussed? Perhaps we do something like `tiledb_query_condition_set_attr_filter` or something like that? And then `tiledb_query_condition_is_null` for the nullable case?